City-of-Helsinki/kukkuu-ui
# fix: profile provider should not reload when tokens are renewed

KK-1152.

The ProfileProvider was reloading the whole sub content every time when
the api tokens were renewed by the OIDC client.

The code is refactored so that the deprecated redux part is moved to a
new hook.

Lots of console-info statements are added for better debugging.

The profile fetching is now in it's own hook and  only the context
handling is now in the provider component.

The helper hook that tries to follow whether the HDS LoginProvider has
finished with the OIDC and API token clients is also refactored and
renamed.

----

How to test that everything works:
1. Basic login: should go to profile page.
2. Login with profile without guardian: should go in registration form.
3. Login to child page: should go to child page through unauthorised route.
4. Make manual reload on profile page and child page and you should stay on the page.
5. Basic log out: should go to home and should see 2 buttons there in hero.
6. Log out the session from another tab: should clear the profile.
7. Check that the token renewal does not reload the page components.
